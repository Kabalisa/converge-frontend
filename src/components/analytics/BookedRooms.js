import React from 'react';
import PropTypes from 'prop-types';
import TableHead from '../helpers/TableHead';
import '../../assets/styles/bookedroom.scss';

const BookedRooms = ({
  pollIcon, moreIcon, bookedRoomText, bookedRoomsList,
}) => (
  <div className="wrap-booked-room">
    <div className="booked-room-header">
      <img src={pollIcon} alt="Pull" />
      <h4>{bookedRoomText}</h4>
      <img src={moreIcon} alt="More" className="" />
    </div>
    <div className="booked-room-list">
      <table>
        <TableHead titles={['Room', 'Meetings', '% Share of All Meetings']} />
        <tbody>
          {bookedRoomsList.map(bookedRoom => (
            <tr key={bookedRoom.id} id="booked-room-body">
              <td>{bookedRoom.name}</td>
              <td>{bookedRoom.meeting}</td>
              <td>{bookedRoom.shareOfAllMeeting}</td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  </div>
);

BookedRooms.propTypes = {
  pollIcon: PropTypes.string.isRequired,
  moreIcon: PropTypes.string.isRequired,
  bookedRoomText: PropTypes.string.isRequired,
  bookedRoomsList: PropTypes.instanceOf(Array).isRequired,
};
export default BookedRooms;
