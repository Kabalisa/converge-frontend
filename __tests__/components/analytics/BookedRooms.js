import React from 'react';
import { shallow, mount } from 'enzyme';
import BookedRooms from '../../../src/components/analytics/BookedRooms';
import bookedRoomsList from '../../../__mocks__/rooms/bookedRoomsList';

describe('BookedRooms component', () => {
  const props = {
    bookedRoomsList,
    pollIcon: '',
    moreIcon: '',
    bookedRoomText: '',
    fetching: false,
    error: null,
    tip: '',
  };
  const rooms = bookedRoomsList.length ? Object.values(bookedRoomsList[0]) : [];
  const meetings = bookedRoomsList.length
    ? Object.values(bookedRoomsList[1])
    : [];
  const meetingShares = bookedRoomsList.length
    ? Object.values(bookedRoomsList[2])
    : [];
  const wrapper = shallow(<BookedRooms {...props} />);
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a single room ', () => {
    expect(rooms).toContain('Asmara');
  });

  it(' should render rooms data in  Rooms table', () => {
    const tableWrapper = mount(
      <BookedRooms
        {...props}
        rooms={rooms}
        meetings={meetings}
        meetingShares={meetingShares}
      />);
    expect(tableWrapper.find('table')
      .children().find('tbody').children()
      .find('tr').length)
      .toBe(5);
  });
});
