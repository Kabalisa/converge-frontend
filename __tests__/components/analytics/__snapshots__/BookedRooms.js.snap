// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookedRooms component should render correctly 1`] = `
<div
  className="wrap-booked-room"
>
  <div
    className="booked-room-header"
  >
    <img
      alt="Pull"
      src=""
    />
    <h4 />
    <img
      alt="More"
      className=""
      src=""
    />
  </div>
  <div
    className="booked-room-list"
  >
    <table>
      <TableHead
        titles={
          Array [
            "Room",
            "Meetings",
            "% Share of All Meetings",
          ]
        }
      />
      <tbody>
        <tr
          id="booked-room-body"
          key="1"
        >
          <td>
            Accra
          </td>
          <td>
            22
          </td>
          <td>
            20%
          </td>
        </tr>
        <tr
          id="booked-room-body"
          key="2"
        >
          <td>
            Antananarivo
          </td>
          <td>
            18
          </td>
          <td>
            13%
          </td>
        </tr>
        <tr
          id="booked-room-body"
          key="3"
        >
          <td>
            Asmara
          </td>
          <td>
            14
          </td>
          <td>
            11%
          </td>
        </tr>
        <tr
          id="booked-room-body"
          key="4"
        >
          <td>
            Brazzaville
          </td>
          <td>
            13
          </td>
          <td>
            9%
          </td>
        </tr>
        <tr
          id="booked-room-body"
          key="5"
        >
          <td>
            Cairo
          </td>
          <td>
            13
          </td>
          <td>
            9
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
